spec/support/shared_entity_specs.rb

```diff
@@ -133,3 +133,74 @@ def verify_signature(pubkey, signature, signed_string)
     end
   end
 end
+
+shared_examples "a JSON Entity" do
+  it "#to_json output matches JSON schema" do
+    json = described_class.new(data).to_json
+    expect(json.to_json).to match_json_schema(:entity_schema)
+  end
+
+  let(:to_json_output) { described_class.new(data).to_json.to_json }
+
+  it "contains described_class property matching the entity class (underscored)" do
+    expect(to_json_output).to include_json(entity_type: described_class.entity_name)
+  end
+
+  it "contains JSON properties for each of the entity properties with the entity_data property" do
+    entity_data = entity_hash_from(data)
+    entity_data.delete(:parent)
+    nested_elements = entity_data.select {|_key, value| value.is_a?(Array) || value.is_a?(Hash) }
+    entity_data.reject! {|_key, value| value.is_a?(Array) || value.is_a?(Hash) }
+
+    expect(to_json_output).to include_json(entity_data: entity_data)
+    nested_elements.each {|key, value|
+      type = described_class.class_props[key]
+      if value.is_a?(Array)
+        data = value.map {|element|
+          {
+            entity_type: type.first.entity_name,
+            entity_data: element
+          }
+        }
+        expect(to_json_output).to include_json(entity_data: {key => data})
+      else
+        expect(to_json_output).to include_json(
+          entity_data: {
+            key => {
+              entity_type: type.entity_name,
+              entity_data: value
+            }
+          }
+        )
+      end
+    }
+  end
+
+  include_examples "JSON is parsable with #from_json" do
+    let(:entity_class) { described_class }
+  end
+end
+
+shared_examples "JSON is parsable with #from_json" do
```

Comment:
Yes, because we test this then for each of the entities and relayables. But since in the specs of relayables we already add signatures in the `data`, maybe add some specs to `relayable_entity_spec.rb` that explicitly tests if signatures/order/additional_data are handled correctly, like we also do for `to/from_xml` (some are already there). But this generic "is parsable" should be OK when it is only used in the `a JSON Entity`.